app/models/page_view.rb

```diff
@@ -6,7 +6,7 @@ class PageView < ApplicationRecord
 
   before_create :extract_domain_and_path
 
-  algoliasearch index_name: "UserHistory", per_environment: true, if: :belongs_to_pro_user? do
+  algoliasearch index_name: "UserHistory", per_environment: true, if: :belongs_to_pro_user?, enqueue: true do
```

Comment:
I suppose that will still trigger delayed jobs with an ActiveRecord object as an argument even in the case of the deletion, which is described in [the docs](https://github.com/algolia/algoliasearch-rails#things-to-consider). I also find their DJ option not useful because of the reasons that I described in the previous comment. 
For now, I suggest using custom jobs as a solution, as in #5063.

Reply:
Talked with @lightalloy and we are going to merge her PR and then use the new job here for this one.